### PR TITLE
fix: npm audit fixes and lock files

### PR DESCRIPTION
**Pull request by joebowbeer**
# Description

* Fixes most npm audit issues in quickstarts and tutorials
* Adds missing package-lock.json files

Also:
* Reduces flakiness of `service_invocation/javascript/http/checkout/index.js`, which was needed to test my changes (see #1061)
* Updates buildclient script in tutorials/distributed-calculator/react-calculator (see comment below)

As it stands, all the affected code installs and runs. Tested in the devcontainer.

However, there are still some audit issues in two react app clients, but to fix them would require updating the react-scripts, which is not in my wheelhouse.

* tutorials/distributed-calculator/react-calculator/client
* tutorials/pub-sub/react-form/client

`npm audit` output includes:

> fix available via `npm audit fix --force`
> Will install react-scripts@3.0.1, which is a breaking change

## Issue reference

Please reference the issue this PR will close: #1063 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] The quickstart code compiles correctly
* [x] You've tested new builds of the quickstart if you changed quickstart code
* [x] You've updated the quickstart's README if necessary
* [x] If you have changed the steps for a quickstart be sure that you have updated the automated validation accordingly. All of our quickstarts have annotations that allow them to be executed automatically as code. For more information see [mechanical-markdown](https://github.com/dapr/mechanical-markdown#readme). For user guide with examples see [Examples](https://github.com/dapr/mechanical-markdown/tree/main/examples#mechanical-markdown-by-example).

### CODE REVIEW

File: tutorials/distributed-calculator/react-calculator/package.json [hunk 1]
```diff
@@ -6,14 +6,14 @@
         "server": "nodemon server.js",
         "dev": "concurrently --kill-others-on-fail \"yarn server\" \"yarn client\"",
         "start": "node server.js",
-        "buildclient": "cd client && npm install && npm install --only=dev --no-shrinkwrap && npm run build"
+        "buildclient": "cd client && npm install && npm run build"
```
Comment (joebowbeer) [inline tutorials/distributed-calculator/react-calculator/package.json:9]: `--only=dev` is not a supported option and `--no-shrinkwrap` is obsolete

Compare to the buildclient script in tutorials/pub-sub/react-form/package.json

**Comment by paulyuk** [comment]
Thank you @joebowbeer - this was in pretty bad need of a dependency check.  Looks great.  I'm trying to ship out of `release-1.14` branch given we're in endgame.  Could you please sync up and fix merge conflicts there?  (just just be package locks fixed up by a new npm install).  Then we are likely good to go and I'll take it for the release.

**Comment by joebowbeer** [comment]
@paulyuk rebased - PTAL

**Review by paulyuk — APPROVED** [review]
LGTM

**Comment by paulyuk** [comment]
just waiting for checks and then we'll merge.  Thank you!

**Pull request merged**